vic/drivers/cesm/src/cesm_interface_c.c

```diff
@@ -105,25 +105,26 @@ vic_cesm_run(vic_clock *vclock)
 
     // advance the clock
     advance_time();
-    assert_time_insync(vclock, &dmy);
+    assert_time_insync(vclock, &dmy_current);
 
     // read forcing data
     vic_force();
 
     // run vic over the domain
-    vic_image_run();
+    vic_image_run(&dmy_current);
 
     // return fields to coupler
     vic_cesm_put_data();
 
     // if output:
-    if (check_write_flag(current)) {
+    if (check_write_flag(current - 1)) {
```

Comment:
Exactly.  This brings up another point.  In general, I'm not sure the best way to tackle the timestep advancing issue. I think it would actually to be best to advance and check at the end of the timestep.  That would allow us to avoid using `current -1` and is more inline with how CESM views the timestamp it passes us (end of the current coupling period).